### ISSUE TITLE
Squiz.NamingConventions.ValidVariableName should not report errors when reading properties from an object

### ISSUE THREAD

**Issue by stof** [PHPCSStandards/PHP_CodeSniffer#1496]
## Describe the bug

Reading a property from an object has to use the property name defined by the class declaration. When that class is a third-party one, the project has no way to use a camelCased name for the property.

### Code sample
```php
function processObject(ThirdPartyClass $object): void
{
    $customerId = $object->customer_id;
}
```

## Expected behavior

`Squiz.NamingConventions.ValidVariableName` should report errors only for class member declarations, not for class member usages.

## Versions (please complete the following information)

|                                          |                                                                                                                |
|-----------------------------|---------------------------------------------------------------------------- |
| Operating System            | Ubuntu                                                           |
| PHP version                      | 8.5.10                                                                                          |
| PHP_CodeSniffer version | 4.0.4                                                                                  |
| Standard                          | Squiz                                                          |
| Install type                       | composer (local) |

## Additional context

This happens for instance when using the Stripe SDK, which uses snake case names for its magic properties.

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `4.x` branch of PHP_CodeSniffer.
